code/modules/recycling/conveyor2.dm

```diff
@@ -123,35 +125,43 @@ GLOBAL_LIST_EMPTY(conveyors_by_id)
 		return FALSE
 	return TRUE
 
-	// machine process
-	// move items to the target location
-/obj/machinery/conveyor/process()
-	if(machine_stat & (BROKEN | NOPOWER))
-		return
-	//If the conveyor is broken or already moving items
-	if(!operating || conveying)
-		return
-	use_power(6)
-	//get the first 30 items in contents
+// returns true if there is at least 1 item on the belt that can be moved
+/obj/machinery/conveyor/proc/check_belt_contents()
+	for(var/atom/movable/AM in loc)
+		if(!AM.anchored)
+			return TRUE
+	return FALSE
+
+// populates `affecting` with the first 30 items in the belt's contents
+/obj/machinery/conveyor/proc/get_belt_contents()
 	affecting = list()
 	var/i = 0
-	for(var/item in loc.contents)
-		if(item == src)
+	for(var/atom/movable/AM in loc)
+		if(AM.anchored) // If it can't be moved, ignore it
 			continue
-		i++ // we're sure it's a real target to move at this point
-		if(i >= MAX_CONVEYOR_ITEMS_MOVE)
+		if(++i >= MAX_CONVEYOR_ITEMS_MOVE)
```

Comment:
In order to do what you are suggesting to do you'll need either of these:
`if(++i > MAX_CONVEYOR_ITEMS_MOVE)`
or
`if(i++>= MAX_CONVEYOR_ITEMS_MOVE)`
Else you'll be moving one less than the `MAX_CONVEYOR_ITEMS_MOVE`
If, say, the maximum is `1`, it will move none.
That is the old logic, not introduced by your PR from what I see.

Reply:
oh yeah, you're right, gonna go with `++i >`

Reply:
That works. Thanks for the fix!